src/button/button.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * JavaScript 'shim' to trigger the click event of element(s) when the space key is pressed.
+ *
+ * Created since some Assistive Technologies (for example some Screenreaders)
+ * will tell a user to press space on a 'button', so this functionality needs to be shimmed
+ * See https://github.com/alphagov/govuk_elements/pull/272#issuecomment-233028270
+ *
+ * Usage instructions:
+ * the 'shim' will be automatically initialised
+ */
+;(function (global) {
+  'use strict'
+
+  var GOVUK_FRONTEND = global.GOVUK_FRONTEND || {}
+  var KEY_SPACE = 32
+
+  GOVUK_FRONTEND.buttons = {
+
+    /**
+    * Add event construct for modern browsers or IE
+    * which fires the callback with a pre-converted target reference
+    * @param {object} node element
+    * @param {string} type event type (e.g. click, load, or error)
+    * @param {function} callback function
+    */
+    addEvent: function (node, type, callback) {
+      if (node.addEventListener) {
+        node.addEventListener(type, function (e) {
+          callback(e, e.target)
+        }, false)
+      } else if (node.attachEvent) {
+        node.attachEvent('on' + type, function (e) {
+          callback(e, e.srcElement)
+        })
+      }
+    },
+
+    /**
+    * Cross-browser character code / key pressed
+    * @param {object} e event
+    * @returns {number} character code
+    */
+    charCode: function (e) {
+      return (typeof e.which === 'number') ? e.which : e.keyCode
+    },
+
+    /**
+    * Cross-browser preventing default action
+    * @param {object} e event
+    */
+    preventDefault: function (e) {
+      if (e.preventDefault) {
+        e.preventDefault()
+      } else {
+        e.returnValue = false
+      }
+    },
+
+    eventHandler: function (e) {
```

Comment:
Would be ace to document this in the same way the other functions are documented.